core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
Sorry, I'm not comfortable with your assumption. Without changing the interface to properly return the executors that were actually removed, this change is too risky.